logstash/check.py

```diff
@@ -0,0 +1,304 @@
+        "logstash.pipeline.events.in": ("gauge", "pipeline.events.in"),
+        "logstash.pipeline.events.out": ("gauge", "pipeline.events.out"),
+        "logstash.pipeline.events.filtered": ("gauge", "pipeline.events.filtered"),
+        "logstash.pipeline.reloads.successes": ("gauge", "pipeline.reloads.successes"),
+        "logstash.pipeline.reloads.failures": ("gauge", "pipeline.reloads.failures")
+    }
+
+    PIPELINE_INPUTS_METRICS = {
+        "logstash.pipeline.plugins.inputs.events.out": ("gauge", "events.out"),
+        "logstash.pipeline.plugins.inputs.events.queue_push_duration_in_millis": ("gauge", "events.queue_push_duration_in_millis")
+    }
+
+    PIPELINE_OUTPUTS_METRICS = {
+        "logstash.pipeline.plugins.outputs.events.in": ("gauge", "events.in"),
+        "logstash.pipeline.plugins.outputs.events.out": ("gauge", "events.out"),
+        "logstash.pipeline.plugins.outputs.events.duration_in_millis": ("gauge", "events.duration_in_millis")
+    }
+
+    PIPELINE_FILTERS_METRICS = {
+        "logstash.pipeline.plugins.filters.events.in": ("gauge", "events.in"),
+        "logstash.pipeline.plugins.filters.events.out": ("gauge", "events.out"),
+        "logstash.pipeline.plugins.filters.events.duration_in_millis": ("gauge", "events.duration_in_millis")
+    }
+
+    def get_instance_config(self, instance):
+        url = instance.get('url')
+        if url is None:
+            raise Exception("A URL must be specified in the instance")
+
+        # Support URLs that have a path in them from the config, for
+        # backwards-compatibility.
+        parsed = urlparse.urlparse(url)
+        if parsed[2] != "":
+            url = "%s://%s" % (parsed[0], parsed[1])
+        port = parsed.port
+        host = parsed.hostname
+
+        custom_tags = instance.get('tags', [])
+        service_check_tags = [
+            'host:%s' % host,
+            'port:%s' % port
+        ]
+        service_check_tags.extend(custom_tags)
+
+        # Tag by URL so we can differentiate the metrics
+        # from multiple instances
+        tags = ['url:%s' % url]
+        tags.extend(custom_tags)
+
+        timeout = instance.get('timeout') or self.DEFAULT_TIMEOUT
+
+        config = LogstashInstanceConfig(
+            service_check_tags=service_check_tags,
+            ssl_cert=instance.get('ssl_cert'),
+            ssl_key=instance.get('ssl_key'),
+            ssl_verify=instance.get('ssl_verify'),
+            tags=tags,
+            timeout=timeout,
+            url=url
+        )
+        return config
+
+    def _get_data(self, url, config, send_sc=True):
+        """ Hit a given URL and return the parsed json
+        """
+        auth = None
+
+        # Load SSL configuration, if available.
+        # ssl_verify can be a bool or a string (http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification)
+        if isinstance(config.ssl_verify, bool) or isinstance(config.ssl_verify, str):
```

Comment:
If we check the type of `ssl_verify` we should print an error when we set `verify` to `None`.